staging/src/k8s.io/component-base/cli/flag/map_string_string.go

```diff
@@ -23,29 +23,35 @@ import (
 )
 
 // MapStringString can be set from the command line with the format `--flag "string=string"`.
-// Multiple flag invocations are supported. For example: `--flag "a=foo" --flag "b=bar"`. If this is desired
-// to be the only type invocation `NoSplit` should be set to true.
-// Multiple comma-separated key-value pairs in a single invocation are supported if `NoSplit`
-// is set to false. For example: `--flag "a=foo,b=bar"`.
+// Multiple flag invocations are supported.
+// For example: `--flag "a=foo" --flag "b=bar"`.
+// Multiple comma-separated key-value pairs in a single invocation are supported,
+// when MapStringStringOptions.DisableBatch=false.
+// For example: `--flag "a=foo,b=bar"`.
 type MapStringString struct {
 	Map         *map[string]string
 	initialized bool
-	NoSplit     bool
+	options     *MapStringStringOptions
+}
+
+// MapStringStringOptions contains options that control how the values are parsed
+type MapStringStringOptions struct {
+	// DisableBatch disables parsing multiple comma-separated key-value pairs
+	// from a single invocation. Instead, the entire string after the = separator
+	// will be parsed as the value. This can be convenient if values contain commas.
+	DisableBatch bool
```

Comment:
How we just be explicit and call it `DisableCommaSeparatedPairs`?

Reply:
Oh that's actually pretty great. Thanks!

Reply:
Updated.